static/index.html

```diff
@@ -1,69 +1,126 @@
+    <meta name="viewport" content="width=device-width">
+    <link rel="stylesheet" type="text/css" href="/resources/style.css" />
+  </head>
+  <body>
+    <h1>mdn-bcd-collector</h1>
+    <p>Data collection service for <a href="https://github.com/mdn/browser-compat-data">mdn/browser-compat-data</a>. This website provides automatic feature detection for Web APIs and CSS properties, to assist in obtaining data for the BCD project.</p>
 
-      dropdown.onchange = function() {
-        var choice = dropdown.options[dropdown.selectedIndex].textContent;
-        button.textContent = 'Run ' + choice;
-      }
+    <div>
+      <button id="start" disabled>Loading tests...</button>
+      <select id="tests" disabled>
+        <option value="">Loading tests...</option>
+      </select>
+      <br />
+      <div id="individualTestOptionsBox" style="display: none;">
+        <input type="checkbox" id="secureContext"> <label for="secureContext">Use HTTPS?</label>
+        <div id="limitScopeBox" style="display: none;">
+        <label for="limitScope">Limit Scope:</label>
+        <select id="limitScope">
+          <option value="">All Scopes</option>
+          <option value="?scope=Window">Window</option>
+          <option value="?scope=Worker">Worker</option>
+          <option value="?scope=ServiceWorker">Service Worker</option>
+        </select>
+      </div>
+    </div>
 
-      button.onclick = function() {
-        var newUrl = dropdown.options[dropdown.selectedIndex].value;
-        if (newUrl) window.location = newUrl;
-      }
-    }
-  }
-}
-getTests();
-</script>
+    <p>Note: this website is in beta; not all feature detection is accurate yet. If you find any errors, please file an issue on the GitHub page.</p>
+    <p>Known caveats:</p>
+    <ul>
+      <li>Detection for features under alternative names is not yet supported</li>
+      <li>In older browsers, some attributes aren't initialized in the API prototypes, resulting in false negatives</li>
+      <li>Some APIs are accessed through other APIs and aren't exposed on their own; custom tests have not been written for them all</li>
+    </ul>
+
+    <p><a href="https://github.com/foolip/mdn-bcd-collector">Source on GitHub</a></p>
+
+    <script src="/resources/json3.min.js"></script>
+    <script>
+    function getDropdownSelection(dropdown) {
+      var index = dropdown.selectedIndex;
+      var choice = dropdown.options[index].value;
+      return [index, choice];
+    };
 
+    function getTests() {
+      var button = document.getElementById('start');
+      var dropdown = document.getElementById('tests');
+      var individualTestOptionsBox = document.getElementById('individualTestOptionsBox');
+      var secureContext = document.getElementById('secureContext');
+      var limitScope = document.getElementById('limitScope');
 
-<p><a href="https://github.com/foolip/mdn-bcd-collector">Source on GitHub</a></p>
-</body>
+      var client = new XMLHttpRequest();
```

Comment:
Since it's within the scope of server-side builds, I decided to actually tackle this, and hook up EJS to build the pages.  All pages -- including the homepage, results, and the tests themselves -- are now built with the EJS templating system!